engine/src/main/java/com/redhat/cloud/notifications/templates/Default.java

```diff
@@ -0,0 +1,71 @@
+package com.redhat.cloud.notifications.templates;
+
+import com.redhat.cloud.notifications.models.EmailSubscriptionType;
+import io.quarkus.qute.CheckedTemplate;
+import io.quarkus.qute.TemplateInstance;
+
+/**
+ * Name needs to be "Default" to read templates from resources/templates/Default
+ *
+ * This class handles the default emails by wrapping the original EmailTemplate.
+ * If the title or body is already supported, it will pass trough.
+ * If the title or body is not supported in the wrapped class AND the type is instant, it will return the default one.
+ * Else it will throw an exception (or reuse the one provided by the call to the wrapped class)
+ * This is only useful for the templates as files flow.
+ *
+ * For templates in database, it provides static methods for getting the TemplateInstance for the title and body.
+ */
+public class Default implements EmailTemplate {
```

Comment:
The latest changes in `TemplateRepository` look good to me 👍

There is however something I don't understand in this class. What is the point of wrapping templates into a `Default` (when the feature is enabled) if the logic that determines [whether the default body/title should be used or not] is located in `EmailSubscriptionTypeProcessor`? Maybe I missed something but it feels like `EmailSubscriptionTypeProcessor` would keep working if you removed the wrapping here.

Reply:
Let me check again, the whole point of this started as a way to centralize that logic on this class, but surely that has diverged from the intention.

Reply:
As said, previously this `Default` also had the logic to decide if we use the original template or the default template. 

I had to move out this logic to take into account the case when only one of  the templates was present (i.e. not mixing default title with original and viceversa).

The logic that remains there is to check for the `isSupported*` methods, it will always support anything that has `INSTANT` regardless of the wrapped class value. This is used to avoid duplicating that logic in `user preferences` and `EmailSubscriptionTypeProcessor`.

Reply:
Ok, I missed a few details, the wrapping is indeed necessary. Thanks for clarifying.

The current code is a bit hard to fully understand at first glance, but the tricky part is only temporary so merging it sounds reasonable to me.

Reply:
Yes - the idea is to get rid of it soon - Actually the part where the "warning" is added to the default body template only happens with in-file templates - as database templates will require both templates to exist.